lldb-eval/eval.cc

```diff
@@ -1184,13 +1244,13 @@ Value Interpreter::EvaluateBinarySubAssign(Value lhs, Value rhs) {
 
   if (lhs.IsPointer()) {
     assert(rhs.IsInteger() && "invalid ast: rhs must be an integer");
-    ret = EvaluateBinarySubtraction(lhs, rhs);
+    ret = EvaluateBinarySubtraction(lhs, rhs, rhs.type());
   } else {
     assert(lhs.IsScalar() && "invalid ast: lhs must be an arithmetic type");
     assert(rhs.type()->IsBasicType() &&
            "invalid ast: rhs must be a basic type");
     ret = CastScalarToBasicType(target_, lhs, rhs.type(), error_);
-    ret = EvaluateBinarySubtraction(ret, rhs);
+    ret = EvaluateBinarySubtraction(ret, rhs, rhs.type());
```

Comment:
Here `ret.type() == rhs.type()`, right (because we just did the cast)? In other places when choosing between two arguments that are the same, we usually pick the first one (kinda like a "convention").

Reply:
Replaced with `ret.type()`.